services/provider/server/server.go

```diff
@@ -832,6 +911,39 @@ func (s *OCSProviderServer) GetStorageClaimConfig(ctx context.Context, req *pb.S
 
 }
 
+func getExternalResourceLabels(kind string, isReplicationEnabled bool, isMirroringEnabled bool, isFlattenMode bool,
```

Comment:
We don't need to differentiate between the Replication and Mirroring flag, we can have one mirroring flag (which will check the cephblockpool) and add labels based on that.

Reply:
From my perspective, there's no other way to handle these flags. Replication is `Vol*RepClass` specific, while Mirroring is aimed for both `Vol*RepClass` and `Vol*SnapClass`.

Reply:
Your replication flag depends on storageClusterPeer, whereas the mirroring flag is derived from CephBlockPool.

Enabling mirroring on the cephblockpool is the responsibility of the storageClusterPeer. Isn't both pointing to the same field eventually?

I think we should add the labels for rbd-based V*RC/V*SC and SC based only on the mirroring field. It is possible that a particular blockpool is not selected for mirroring from StorageClusterPeer CR. See [here](https://github.com/red-hat-storage/ocs-operator/blob/main/api/v1/storageclusterpeer_types.go#L50-L53)

I was going through the PR and I am not able to find the labels applied for CephFS-based V*SC and SC. Do we not need them?

Reply:
>I am not aware of such thing. How does it fetch this flag exactly? Is it merged already?

This was discussed in the design discussion. Each StorageClusterPeer will enable mirroring on the cephblockpool based on the label selector.

For rbd we can use the mirroring field on the cephblockpool to add ramenRelated labels, we don't need another replication flag explicitly.

Reply:
I think you're misunderstanding. I am using the Mirroring flag solely for the StorageID label. It's unrelated to RepID.

Reply:
Why do we need two flags for the same set of ramen labels? From what I understand we are creating a differentiation which we don't need. It will just complicate the things. 

As an example what if we have 2 cephblockpool a and b the storageclusterpeer is configured only for blockpool a.

There are two clients 1 and 2. 1 uses blockpool a, 2 uses blockpool b.

If we keep the flags separate,

For client 1 we will enable both replication and storage id label

For client 2 we will enable only replication id label

In the second case we don't require any ramen related labels to be added as that blockpool is not configured for mirroring. 

Hence we don't need to check for both storageclusterpeer and cephblockpool. Checking only the cephblockpool will work

Reply:
> Each StorageClusterPeer will enable mirroring on the cephblockpool based on the label selector.

It doesn't make much sense to me to retrieve this flag by checking each SCP instead of a single CBP.

Reply:
For rbd,
StorageClass will created from a radosnamesapce on a blockpool. From my perspective to populate the mirroring and replication flag, we should check if the mirroring is enabled for radosnamespace [spec.mirroring ](https://github.com/rook/rook/pull/14701)

For cephfs,
Since there is not need to perform any actions for this on ceph level unlike rbd, we should enable the mirroring and replication flag for this as soon as we have a storageClusterPeer present